ros2cli/ros2cli/node/daemon.py

```diff
@@ -119,6 +121,20 @@ def spawn_daemon(args, timeout=None, debug=False):
       `False` if it was already running.
     :raises: if it fails to spawn the daemon.
     """
+    # Make file handles(except for 0, 1, 2) not inheritable for the incoming daemon process
+    for i in range(3, 255):
```

Comment:
i guess this is really enough but we would want to ask the system for this?

https://docs.python.org/3/library/resource.html#resource.getrlimit

```python
import resource

soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)

for i in range(3, soft):
...
```

Reply:
Thanks for your suggestion.